File: app/src/main/java/de/tum/in/tumcampus/models/CalendarRowSet.java
package de.tum.in.tumcampus.models;

import java.util.List;

import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

/**
 * This class is dealing with the deserialization of the output of TUMOnline to
 * the method "sucheLehrveranstaltungen" or "eigeneLehrveranstaltungen".
 * 
 * @author Daniel Mayr
 * @see LecturesSearchRow
 * @see http://simple.sourceforge.net/download/stream/doc/tutorial/tutorial.php
 * @review Thomas Behrens
 */
@Root(name = "events")
public class CalendarRowSet {

	@ElementList(inline = true, required = false)
	private List<CalendarRow> kalendarList;

	public List<CalendarRow> getKalendarList() {
		return kalendarList;
	}

	public void setKalendarList(List<CalendarRow> kalendarList) {
		this.kalendarList = kalendarList;
	}

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/model/CalendarRowSet.kt
package de.tum.`in`.tumcampusapp.component.tumui.calendar.model

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Xml

/**
 * This class is dealing with the deserialization of the output of TUMOnline to
 * the method "sucheLehrveranstaltungen" or "eigeneLehrveranstaltungen".
 *
 * @see LecturesSearchRow
 *
 * @see [SimpleXML tutorial](http://simple.sourceforge.net/download/stream/doc/tutorial/tutorial.php)
 */
@Xml(name = "events")
data class CalendarRowSet(@Element var kalendarList: List<CalendarRow>? = null) // TODO: Rename variable
